pkg/block/s3/adapter.go

```diff
@@ -412,8 +411,11 @@ func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer,
 		log.WithError(err).Error("could not pre-sign request")
 		return "", time.Time{}, err
 	}
+
+	// in case the credentials can expire, we need to use the earliest expiry time
+	// we assume that session expiry window is used and adjust the expiry time accordingly
 	if captureExpiresPresigner.CredentialsCanExpire && captureExpiresPresigner.CredentialsExpireAt.Before(expiry) {
-		expiry = captureExpiresPresigner.CredentialsExpireAt
+		expiry = captureExpiresPresigner.CredentialsExpireAt.Add(a.sessionExpiryWindow)
```

Comment:
**PLEASE** add a comment before this.  I did not understand the logic until you explained it to me!
Perhaps something like
> ```go
> // AWS Go SDK v2 stores the time to renew credentials in `CredentialsExpireAt`.  This is
> // a.sessionExpiryWindow before actual credentials expiry.
> ```